centralized-sampling-tests/sample-apps/spring-boot/src/main/java/com/amazon/sampleapp/AppController.java

```diff
@@ -0,0 +1,105 @@
+  @GetMapping(value = "/getSampled")
+  @ResponseBody
+  public int getSampled(
+      @RequestHeader("user") String userAttribute,
+      @RequestHeader("service_name") String name,
+      @RequestHeader("required") String required,
+      @RequestHeader("totalSpans") String totalSpans) {
+
+    Attributes attributes =
+        Attributes.of(
+            AttributeKey.stringKey("http.method"), "GET",
+            AttributeKey.stringKey("http.url"), "http://localhost:8080/getSampled",
+            AttributeKey.stringKey("user"), userAttribute,
+            AttributeKey.stringKey("http.route"), "/getSampled",
+            AttributeKey.stringKey("required"), required,
+            AttributeKey.stringKey("http.target"), "/getSampled");
+    return getSpans(name, totalSpans, attributes);
+  }
+
+  // Post endpoint for /getSampled that requires three header values for user, service_name, and
+  // required
+  // Returns the number of times a span was sampled out of the creation of 1000 spans
+  @PostMapping("/getSampled")
+  @ResponseBody
+  public int postSampled(
+      @RequestHeader("user") String userAttribute,
+      @RequestHeader("service_name") String name,
+      @RequestHeader("required") String required,
+      @RequestHeader("totalSpans") String totalSpans) {
+
+    Attributes attributes =
+        Attributes.of(
+            AttributeKey.stringKey("http.method"), "POST",
+            AttributeKey.stringKey("http.url"), "http://localhost:8080/getSampled",
+            AttributeKey.stringKey("user"), userAttribute,
+            AttributeKey.stringKey("http.route"), "/getSampled",
+            AttributeKey.stringKey("required"), required,
+            AttributeKey.stringKey("http.target"), "/getSampled");
+    return getSpans(name, totalSpans, attributes);
+  }
+
+  // Get endpoint for /importantEndpoint that requires three header values for user, service_name,
+  // and required
+  // Returns the number of times a span was sampled out of the creation of 1000 spans
+  @GetMapping("/importantEndpoint")
+  @ResponseBody
+  public int importantEndpoint(
+      @RequestHeader("user") String userAttribute,
+      @RequestHeader("service_name") String name,
+      @RequestHeader("required") String required,
+      @RequestHeader("totalSpans") String totalSpans) {
+    Attributes attributes =
+        Attributes.of(
+            AttributeKey.stringKey("http.method"), "GET",
+            AttributeKey.stringKey("http.url"), "http://localhost:8080/importantEndpoint",
+            AttributeKey.stringKey("http.route"), "/importantEndpoint",
+            AttributeKey.stringKey("required"), required,
+            AttributeKey.stringKey("user"), userAttribute,
+            AttributeKey.stringKey("http.target"), "/importantEndpoint");
+    return getSpans(name, totalSpans, attributes);
+  }
+
+  public static int getSpans(String name, String totalSpans, Attributes attributes) {
+    int numSampled = 0;
+    int spans = Integer.parseInt(totalSpans);
```

Comment:
why Integer?

Reply:
Thought you meant to not set as a string in a comment up above, works the same this way without an extra cast. Http calls auto make the response a string so nothing changes